File: tests/fixtures/sample.ts
import def from "./mod-a";
import type { T } from "./types";
const cfg = require("mod-c");

/** Greet the user. */
export function greet(name: string): string {
  return "hi " + name;
}

export class Greeter {
  /** Greet doc. */
  greet(name: string): string {
    return name;
  }
}

const double = (x: number): number => x * 2;
const d = await import("./dyn");

File: tests/fixtures/sample.cpp
#include <vector>
#include "foo.h"

// Greeter class.
class Greeter {
public:
    Greeter();
    // Greet a name.
    std::string greet(const std::string& name);
};

// Double a value.
template <typename T>
T double_it(T x) {
    return x * 2;
}
